src/modules/application/controllers/controllerEdit.js

```diff
@@ -0,0 +1,68 @@
+import { verifyToken } from "../../../utilities/jsonWebToken.js";
+import { databasePrisma } from "../../../prismaClient.js";
+
+export const handleEdit = async function (req) {
+  const { id: applicationId } = req.params;
+  const { applicant, listing, coverLetter, company } = req.body;
+
+  const token = req.headers.authorization;
+
+  let readyToken = token;
+  if (!token) {
+    return Promise.resolve({
+      status: 401,
+      message: "User has to be authenticated to make this request",
+    });
+  } else if (token.includes("Bearer")) {
+    readyToken = token.slice(7);
+  }
+
+  let verified;
+
+  if (readyToken !== undefined) {
+    try {
+      verified = verifyToken(readyToken);
+    } catch (error) {
+      return Promise.resolve({ status: 401, message: "Auth token not valid." });
+    }
+  }
+
+  if (verified) {
+    try {
+      const result = await databasePrisma.application.update({
+        where: { id: applicationId },
+        data: {
+          applicant: { connect: { id: applicant } },
+          listing: { connect: { id: listing } },
+          company: { connect: { id: company } },
+          coverLetter,
+        },
+      });
+
+      result.response = "Your Application is updated successfully";
+
+      return { status: 200, data: result };
+    } catch (error) {
+      if (!error.status) {
+        if (error.meta !== undefined) {
+          return Promise.resolve({
+            status: 409,
+            message: `You've already updated your application`,
```

Comment:
This is a confusing response to receive, why would I have already updated my application?, surely it should refer to an error when attempting to update the database?

Reply:
Fixed